### DISCUSSION TITLE
Without `from` option PostCSS could generate wrong source map and will not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning.

### DISCUSSION THREAD

**Discussion by kud** [postcss/postcss#1621] [Q&A]
Hello,

In my custom webpack configuration, I've got this error when I use postcss.

```js
      {
        test: /\.css$/,
        exclude: /\.module\.css$/,
        use: [
          MiniCssExtractPlugin.loader,
          {
            loader: "css-loader",
            options: {
              modules: false,
              importLoaders: 1,
              import: true,
            },
          },
          {
            loader: "postcss-loader",
            options: {
              postcssOptions: {
                plugins: [
                  ["autoprefixer"],
                  ["postcss-custom-media", { importFrom: [{ customMedia }] }],
                ],
              },
            },
          },
        ],
      },
```

but I don't know what to specify about `from`.

The doc says:

> from: the input file name (most runners set it automatically).

but it seems to be not my case.

Do you have any idea how to fix this?

Thank you.

**Comment by ai** [comment]
`from` have to be set only if you use low-level PostCSS API.

For webpack, you do not need `from` option in loader options (`postcss-loader` set `from` automatically).

A few ideas why you have this warning:
1. You may use very old `postcss-loader`
2. It may be generated by some option tool, which use `postcss.parse()` or `postcss().process()` inside. Check `postcss-custom-media` sources and `npm ls`

**Reply by kud** [reply]
Thank you so so much @ai for your fast answer.

Current versions:

```
    "postcss-custom-media": "8.0.0",
    "postcss-loader": "6.1.1",
    "postcss-reporter": "7.0.2",
```

I'll check asap the custom-media sources. :)
